clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -3443,9 +3444,26 @@ class SyclKernelIntHeaderCreator : public SyclKernelFieldHandler {
   }
 
   bool handlePointerType(FieldDecl *FD, QualType FieldTy) final {
-    addParam(FD, FieldTy,
-             ((StructDepth) ? SYCLIntegrationHeader::kind_std_layout
-                            : SYCLIntegrationHeader::kind_pointer));
+    QualType PointeeTy = FieldTy->getPointeeType();
+    unsigned int Dim = 0;
+    ASTContext &Context = SemaRef.getASTContext();
+    // VLAs are encoded as (Dim << 24) | Size
+    // where Dim is the number of VLA dimensions and
+    // Size is the size of the underlying type
```

Comment:
Oh, so we don't keep track of what extents are used for each dimension?

Reply:
Thanks for asking this.   I have redone the comment to indicate that there are as many kind_std_layout entries preceding the vla entry as the number of dimensions; and each of those describe the extent of their respective dimension.